### DISCUSSION TITLE
Requesting Feedback on Specific Unit Names for Oil and Gas Industry

### DISCUSSION THREAD

**Discussion by rami-alloush** [qudt/qudt-public-repo#735] [Q&A] [answered]
Hi,
We are working on adding units that are used heavily in the oil and gas industry. The `Thousand Cubic Feet` is a base unit that gets used heavily in derivative units like `Thousand Cubic Feet Per Minute` and `Thousand Cubic Feet Per Day`.

The plan was to create a unit with qname `KiloFT3` to reflect that, until I read the example below and realized that this would mean something different.
> Example: KiloM3 means (Kilometre)<sup>3</sup>, or cubic kilometres, not Kilo(M<sup>3</sup>), or 1000 cubic metres, because Prefixes precede Power. Thus, the conversionMultiplier to cubic metres is 10<sup>9</sup> rather than 10<sup>3</sup>.

My question is: is there a way to add `Thousand Cubic Feet` to the ontology (that would equal to `28.31684659‬ M3` with a `conversionMultiplier: 28.31684659‬`)

Thanks

**Comment by jhodgesatmb** [comment]
﻿If I understand you correctly you want a cubic foot unit and a cubic meter unit so that the conversion from cubic foot to cubic meter would be 28.31684659. Then adding a kilo prefix the multiplier would be 1000 times the existing multiplier of 28.31684659, or 28,316.84659. Then the problem would be unique naming. Is this a correct read of your request?Jack Hodges, Ph.D.Arbor StudiosOn Aug 10, 2023, at 3:34 PM, Rami Alloush ***@***.***> wrote:﻿
Hi,
We are working on adding units that are used heavily in the oil and gas industry. The Thousand Cubic Feet is a base unit that gets used heavily in derivative units like Thousand Cubic Feet Per Minute and Thousand Cubic Feet Per Day.
The plan was to create a unit with qname KiloFT3 to reflect that, until I read the example below and realized that this would mean something different.

Example: KiloM3 means (Kilometre)3, or cubic kilometres, not Kilo(M3), or 1000 cubic metres, because Prefixes precede Power. Thus, the conversionMultiplier to cubic metres is 109 rather than 103.

My question is: is there a way to add Thousand Cubic Feet to the ontology (that would equal to 28.3168466 M3 with a conversionMultiplier: 28.31684659‬)
Thanks

**Comment by steveraysteveray** [comment]
I thought this issue might eventually rear its ugly head! I have two thoughts I'd be interested in hearing reactions to:
1. Use explicit parentheses to show the distinct interpretation, as in unit:Kilo(FT3) and unit:Kilo(FT3)-PER-MIN. This seems the most intuitive solution, but "(" and ")" are reserved characters that may not be universally interpreted as part of a URI. This is less of a problem than it used to be, but for true universality we would have to say unit:Kilo%28FT3%29-PER-MIN, which I'm not too happy about.
2. Use "Of" as in unit:KiloOfFT3 and unit:KiloOfFT3-PER-MIN. More universal, but less intuitive. 
3. There are variations we could consider, such as unit:Kilo-OF-FT3...

Thoughts? Votes?

**Comment by rami-alloush** [comment]
Hi Steve,
Thank you for your comprehensive feedback. I'm inclined to use `unit:Kilo-OF-FT3`, it seems to be the most universal and easier to explain and document.
I'll try to get more opinions and votes from other colleagues. 

@nonodename any thoughts?

**Reply by nonodename** [reply]
Sorry, late to the party. I agree with @steveraysteveray  that better to avoid URL reserved characters. so OF makes most sense and '-' character aids readability and is used elsewhere in QUDT.

**Comment by jhodgesatmb** [comment]
The problem here is that they want a unit for FT3 and M3, which means there could be any number of them, and they want the conversion multiplier to follow the unit, not the convention we use now, which would affect all conversion multipliers. The for er requires that we give the new unit a new name, not just some kind of qualifier for the prefix. In fact, if we solve the unit name issue we solve the prefix issue. I do not have a suggestion at this time for the conversion multiplier problem.JackSent from my iPadOn Aug 11, 2023, at 6:13 AM, steveraysteveray ***@***.***> wrote:﻿
I thought this issue might eventually rear its ugly head! I have two thoughts I'd be interested in hearing reactions to:

Use explicit parentheses to show the distinct interpretation, as in unit:Kilo(FT3) and unit:Kilo(FT3)-PER-MIN. This seems the most intuitive solution, but "(" and ")" are reserved characters that may not be universally interpreted as part of a URI. This is less of a problem than it used to be, but for true universality we would have to say unit:Kilo%28FT3%29-PER-MIN, which I'm not too happy about.
Use "Of" as in unit:KiloOfFT3 and unit:KiloOfFT3-PER-MIN. More universal, but less intuitive.
There are variations we could consider, such as unit:Kilo-OF-FT3...

Thoughts? Votes?

**Comment by steveraysteveray** [comment]
I'm not sure I understand your concern, @jhodgesatmb. I don't believe they are requesting a new unit for FT3. They are requesting a new URI naming convention for thousands of cubic feet (as opposed to cubic KiloFT). And accordingly, these new units would have different conversion multipliers using our current conventions.

**Reply by rami-alloush** [reply]
That's totally correct.

**Comment by jhodgesatmb** [comment]
Our naming convention is clear. We apply prefixes to units and exponents
are applied to the unit before the prefix is applied. If we want to do what
the poster wants then we need an explicit unit for KiloFT or KiloM (and any
such units) so that the exponent applies to that and not by our naming
convention. Why not a KFT and KM unit? They do not exist in the vocabulary
at the present time, but there is no telling whether this strategy would
work for all units and prefixes.

I misunderstood the conversion multiplier as I think our current conversion
multiplier is what they said they said they wanted the value to be (i.e.,
1000 times the conversion from FT to M).

On Fri, Aug 11, 2023 at 7:57 AM steveraysteveray ***@***.***>
wrote:

> I'm not sure I understand your concern, @jhodgesatmb
> <https://github.com/jhodgesatmb>. I don't believe they are requesting a
> new unit for FT3. They are requesting a new URI naming convention for
> thousands of cubic feet (as opposed to cubic KiloFT). And accordingly,
> these new units would have different conversion multipliers using our
> current conventions.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/discussions/735#discussioncomment-6702713>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWLIR6VDANFXWFUNLSDXUZB67ANCNFSM6AAAAAA3MELAEE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jack

**Reply by steveraysteveray** [reply]
Our naming convention is indeed clear, but it's the exact converse of what you stated, as quoted at the beginning of this thread and in our wiki. We apply the prefix first, and then the exponent. Secondly, they are not asking for units for KiloFT. They want thousands of cubic feet as I stated in my last message.

I offered a few options as solutions. Do you have any thoughts on those?

**Comment by rami-alloush** [comment]
Hi @steveraysteveray,
I believe we agree to move forward with `-OF-` solution.
I'll work on creating a PR for a new unit named `unit:Kilo-OF-FT3`. Do the Submission Guidelines need to be updated first?
Thanks

**Comment by jhodgesatmb** [comment]
My apologies to all for my [very] embarrassing misinterpretation of our
qname naming conventions which I in fact helped to codify. This approach
will need to be approved by the board before it is implemented. I seem to
recall somewhere in this thread that the usage of "-OF-" has some precedent
in QUDT (or perhaps it was elsewhere). At the present time I believe that
there is only one usage of hyphens in QUDT qnames, and that is for the
"PER" qualifier that separates qname elements in the numerator from those
in the denominator. We use underscores for all qualifiers and I am
wondering if perhaps the usage of "_OF_" was considered over "-OF-". It is
difficult to find an approach that is unique, doesn't confuse or conflict
with the existing qname naming convention, and would be readily understood
by the community.

On Mon, Aug 14, 2023 at 2:31 PM Rami Alloush ***@***.***>
wrote:

> Hi @steveraysteveray <https://github.com/steveraysteveray>,
> I believe we agree to move forward with -OF- solution.
> I'll work on creating a PR for a new unit named unit:Kilo-OF-FT3. Do the
> Submission Guidelines need to be updated first?
> Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/discussions/735#discussioncomment-6725009>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWMTYZGJYBDEEK53OETXVKKK7ANCNFSM6AAAAAA3MELAEE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jack

**Reply by steveraysteveray** [reply]
We use hyphens in all of our compound units (e.g., unit:BTU_TH-FT-PER-FT2-HR-DEG_F). Take a quick look at [https://www.qudt.org/doc/DOC_VOCAB-UNITS.html](https://www.qudt.org/doc/DOC_VOCAB-UNITS.html). They are used to separate two entities that should be multiplied together. In that sense, using -OF- follows that spirit of multiplication. We are not qualifying the unit the way our other qualifiers do.

**Comment by jhodgesatmb** [comment]
I may be mistaken about hyphens and compound units but if the meaning of hyphen is ‘multiplication’ isn’t the usage of “-OF-“ going to compromise that? I am not saying that “_OF_” wouldn’t be a compromise as I don’t know if there is a way to do this that doesn’t.Jack Hodges, Ph.D.Arbor StudiosOn Aug 15, 2023, at 6:26 AM, steveraysteveray ***@***.***> wrote:﻿
We use hyphens in all of our compound units (e.g., unit:BTU_TH-FT-PER-FT2-HR-DEG_F). Take a quick look at https://www.qudt.org/doc/DOC_VOCAB-UNITS.html. They are used to separate two entities that should be multiplied together. In that sense, using -OF- follows that spirit of multiplication. We are not qualifying the unit the way our other qualifiers do.

**Comment by steveraysteveray** [comment]
I agree, we would not be multiplying "OF" by the unit. The strict way would be to simply use the hyphen, like:

Kilo-FT3, which would be thousands of cubic feet, vs. KiloFT3 for "thousand feet" cubed. I think the difference in these two URIs is too subtle, and would be a source of much confusion. Hence the suggestion of making it clearer, with Kilo-OF-FT3.

**Reply by rami-alloush** [reply]
I didn't think of `Kilo-` directly without the `OF`. This seems to be the cleanest solution, and it complies with all existing rules (no need for Guidelines updates). I understand the subtle difference in URIs, but all the URIs are already very meticulous and case sensitive.
I feel like voting for `Kilo-` over `Kilo-OF-`.

**Comment by jhodgesatmb** [comment]
The one thing that is really clear is the 'English' (i.e., natural language
in the U.S.) meaning of "-OF-". In that respect the suggested solution is a
good one and always was. My concerns are only as previously stated. I like
the Kilo-FT3 solution because it doesn't add the 'multiply' confusion and
in fact works with it. Would that work generally?

On Tue, Aug 15, 2023 at 8:11 AM steveraysteveray ***@***.***>
wrote:

> I agree, we would not be multiplying "OF" by the unit. The strict way
> would be to simply use the hyphen, like:
>
> Kilo-FT3, which would be thousands of cubic feet, vs. KiloFT3 for
> "thousand feet" cubed. I think the difference in these two URIs is too
> subtle, and would be a source of much confusion. Hence the suggestion of
> making it clearer, with Kilo-OF-FT3.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/discussions/735#discussioncomment-6731560>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWPPDMJAC5VAOY4X6G3XVOGQ7ANCNFSM6AAAAAA3MELAEE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jack

**Comment by steveraysteveray** [comment]
I'm starting to hear a consensus. Maybe when the remainder of our Board weighs in, we could adopt the Kilo- solution. It's not going to be a common occurrence, so maybe it's worth the risk of confusion.

**Comment by jhodgesatmb** [comment]
As long as there is no conflict I am fine with this solution approach.

On Tue, Aug 15, 2023 at 11:54 AM steveraysteveray ***@***.***>
wrote:

> I'm starting to hear a consensus. Maybe when the remainder of our Board
> weighs in, we could adopt the Kilo- solution. It's not going to be a common
> occurrence, so maybe it's worth the risk of confusion.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/qudt/qudt-public-repo/discussions/735#discussioncomment-6733392>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AATQRWM5ELQO3DISGIOWHODXVPAURANCNFSM6AAAAAA3MELAEE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Jack

**Comment by steveraysteveray** [comment, marked as answer]
After much discussion, we are all agreed to the use of a prefix followed by a hyphen to meet this need. So, to summarize:

KiloFT3 means (1000 feet) raised to the power 3, i.e. a cube 1000 feet on a side, or 10**9 cubic feet.

Kilo-FT3 means 1000 cubic feet, or 10**3 cubic feet.

The [Unit Submission Guidelines](https://github.com/qudt/qudt-public-repo/wiki/Unit-Vocabulary-Submission-Guidelines) have been updated to reflect this new expressive power (see Rule 6 for an example).

**Reply by rami-alloush** [reply]
Perfect! Thank you.
I’ll get the PR for ‘Kilo-FT3’ out shortly and then we can take it from there. Thanks
